### PR TITLE
Save additional metadata

### PR DESCRIPTION
**Pull request by adamltyson**
Atlas creators can now save additional information into `metadata.json` by passing an optional `additional_metadata` dict to `wrapup_atlas_from_data`. This dict will be joined to the existing metadata dict, and saved to `metadata.json`. The readme is unaffected.

This should be backwards compatible, as `bg_atlasapi.BrainGlobeAtlas.metadata` is just the dict loaded from file.

### CODE REVIEW

**Comment by adamltyson** [comment]
The syntax I've used is Python 3.9 and above only, but I think that's fine as this won't be used in any other workflows.

File: bg_atlasgen/metadata_utils.py [hunk 1]
```diff
@@ -129,10 +131,12 @@ def create_metadata_files(dest_dir, metadata_dict, structures, root_id):
     :param uncompr_atlas_path: path to uncompressed atlas folder
     :param metadata_dict: dict with atlas metadata
     :param structures: list of dictionaries with structures hierarchical info
+    :param additional_metadata: Dict to add to atlas metadata
     """
     # write metadata dict:
     with open(dest_dir / descriptors.METADATA_FILENAME, "w") as f:
-        json.dump(metadata_dict, f)
+        # only save additional metadata to json, don't include in readme
+        json.dump(metadata_dict | additional_metadata, f)
```
Comment (vigji) [inline bg_atlasgen/metadata_utils.py:139]: is this the bit that is python 3.9 only? Is it really worth asking developers to upgrade atlasgen env to 3.9?

**Review by vigji — COMMENTED** [review]
Not sure the python 3.9-only syntax is worth it. but otherwise looks good. Maybe we could take the chance of this PR to also include the "brainglobe developer" name(s) for credit to people porting an atlas to brainglobe!

**Comment by adamltyson** [comment]
> Not sure the python 3.9-only syntax is worth it. but otherwise looks good

Happy to be outvoted on this, but my preference is that we always try to move to the newest Python version. Numpy will stop supporting Python <3.8 at the end of the year. We should try harder with the Atlas API, but for atlasgen, how often are people (other than us) going to use it, once/twice a year?

**Comment by adamltyson** [comment]
@FedeClaudi do you have preference about how to merge two dicts? I can change it to `json.dump({**metadata_dict, **additional_metadata}, f)` for Python <3.9 compatibility (although this will fail if the keys of `additional_metadata` are not all strings). Writing a new function rather than adopting Python 3.9 seems to be an unnecessary headache.

**Comment by vigji** [comment]
I mean, probably I should just keep up with current times, I don't mind the syntax and you definitively have good points on how it is useful.

What about the other point? Implementing in this PR the new key on the credits for the person responsible for the BrainGlobe porting?

**Comment by adamltyson** [comment]
Yep this is probably a good time to add this, I should be able to do that this week.

**Comment by FedeClaudi** [comment]
Yeah, I think Adam makes a good point. The atlas gene environment is quite lightweight anyways so creating a new one is fairly painless, and it wouldn't be a weekly occurrence for anybody.

**Comment by adamltyson** [comment]
So after all that, I realised that `allensdk` effectively doesn't support Python 3.9 (due to pinned dependencies), so I've gone with the  `json.dump({**metadata_dict, **additional_metadata}, f)` syntax for now. Longer term, I'd like to move away from that, but we're stuck with `allensdk` for now.

I've also added an (optional) metadata entry for `atlas_packager`. We can add the `atlas.citation` attribute to the Atlas API I think.

@FedeClaudi, @vigji - take a look, I'm happy to merge when you are.

**Review by vigji — APPROVED** [review]

**Pull request merged**